packages/components/src/Chips/InternalChipDismissible/__tests__/KeyboardNavigation.test.tsx

```diff
@@ -0,0 +1,252 @@
+      expect(firstChip).toHaveFocus();
+    });
+
+    it("should not navigate past first chip with left arrow", async () => {
+      // Focus the first chip
+      const firstChip = getByChipLabelText("first");
+      await userEvent.tab();
+      expect(firstChip).toHaveFocus();
+
+      // Try to go before first chip (should stay on first)
+      await userEvent.keyboard("{ArrowLeft}");
+      expect(firstChip).toHaveFocus();
+    });
+
+    it("should not navigate past Add button with right arrow", async () => {
+      // Navigate to Add button
+      await userEvent.tab();
+      await userEvent.keyboard("{ArrowRight}");
+      await userEvent.keyboard("{ArrowRight}");
+      await userEvent.keyboard("{ArrowRight}");
+
+      const addButton = screen.getByRole("button", { name: "Add" });
+      expect(addButton).toHaveFocus();
+
+      // Try to go past Add button (should stay on Add)
+      await userEvent.keyboard("{ArrowRight}");
+      expect(addButton).toHaveFocus();
+    });
+  });
+
+  describe("Focus Management When Deleting Chips", () => {
+    describe("When clicking remove button", () => {
+      it("should focus the previous chip when removing a non-first chip", async () => {
+        render(<MultipleChipsComponent />);
+
+        // Find the middle chip's remove button and click it
+        const middleChip = getByChipLabelText("middle");
+        const removeButton = getRemoveButton(middleChip);
+        await userEvent.click(removeButton);
+
+        // After deletion, the first chip should have focus
+        await waitFor(() => {
+          const firstChip = getByChipLabelText("first");
+          expect(firstChip).toHaveFocus();
+        });
+      });
+
+      it("should focus the next chip when removing the first chip", async () => {
+        render(<MultipleChipsComponent />);
+
+        // Find the first chip's remove button and click it
+        const firstChip = getByChipLabelText("first");
+        const removeButton = getRemoveButton(firstChip);
+        await userEvent.click(removeButton);
+
+        // After deletion, the new first chip (originally the "middle" chip) should have focus
+        await waitFor(() => {
+          const newFirstChip = getByChipLabelText("middle");
+          expect(newFirstChip).toHaveFocus();
+        });
+      });
+
+      it("should focus the Add button when removing the last remaining chip", async () => {
+        render(<SingleChipComponent />);
+
+        // Find the only chip's remove button and click it
+        const onlyChip = getByChipLabelText("only");
+        const removeButton = getRemoveButton(onlyChip);
+        await userEvent.click(removeButton);
+
+        // After deletion, the Add button should have focus
+        await waitFor(() => {
+          const addButton = screen.getByRole("button", { name: "Add" });
+          expect(addButton).toHaveFocus();
+        });
+      });
+    });
+
+    describe("When using keyboard Delete/Backspace", () => {
```

Comment:
I do like having coverage for deletion VS removal

not sure why we were making a distinction between the two previously. I wonder if that was intentional because it seems very strange.